### PR TITLE
Feat: initial testing setup

### LINKED ISSUE #6: Initial testing setup
Depends on #5 

- [ ] New dependency type `test` in `pyproject.toml`
- [ ] Add dependencies

      "coverage",
      "pytest",
      "pytest-django",
      "pytest-mock",
      "pytest-socket",

- [ ] Create `tests/pytest` directory
- [ ] Create a test file and a test that requests a `/healthcheck` endpoint from the app and asserts a 200 response code and "Healthy" response text
- [ ] Implement this endpoint in the Django app so the test passes
- [ ] Create a GitHub Actions workflow to run pytests on push and pull_request, upload coverage report, comment w/coverage report

### PR DESCRIPTION
**Pull request by lalver1**
Closes #6

This PR
- adds test dependencies
- configures VS Code for pytest
- creates a test file and a test that requests a `/healthcheck` endpoint from the app and asserts a 200 response code and "Healthy" response text. The endpoint is implemented as Django middleware.
- create a GitHub Actions workflow to run pytests on push and pull_request, upload coverage report, comment w/coverage report

~The `Pytest` workflow is failing probably due to a pending `python-coverage-comment-action-data` branch.~

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -34,6 +42,12 @@ line-length = 127
 target-version = ['py312']
 include = '\.pyi?$'
 
+[tool.pytest.ini_options]
+DJANGO_SETTINGS_MODULE = "pems.settings"
+markers = [
+    "request_path: use with session_request to initialize with the given path",
```
Comment (thekaveman) [inline pyproject.toml:48]: Do we need this?

File: tests/pytest/__init__.py
Comment (thekaveman) [inline tests/pytest/__init__.py:1]: I think we want a `conftest.py` file in here, to make sure we turn off socket connections for unit tests (e.g. don't allow unit tests to go out to the internet).

```python
from pytest_socket import disable_socket

def pytest_runtest_setup():
    disable_socket()
```

**Review by thekaveman — CHANGES_REQUESTED** [review]

**Comment by thekaveman** [comment]
> The Pytest workflow is failing probably due to a pending python-coverage-comment-action-data branch.

I feel like it should be able to create that branch? I can't remember what we did in the past, but I'm pretty sure I never created the branch.

This is probably related to GitHub Actions permissions somehow...

File: pyproject.toml (hunk 1 above)
Comment (lalver1) [reply pyproject.toml]: Thanks for catching this, we actually don't need `markers` so I removed it.

File: tests/pytest/__init__.py
Comment (lalver1) [reply tests/pytest/__init__.py:1]: Yes! Added it in 9f8ab8f7476f29466a1fb19e17a5347c6a378aac

**Comment by lalver1** [comment]
Ah, I had forgotten to configure `coverage` in `pyproject.toml`, looks like the GitHub Actions workflow runs now 😅

**Review by thekaveman — APPROVED** [review]
Looks great! I was able to run the tests locally in the devcontainer too.

**Pull request merged**